hazelcast/src/test/java/com/hazelcast/scheduledexecutor/ScheduledExecutorServiceBasicTest.java

```diff
@@ -155,7 +177,34 @@ public void capacity_whenDefault() {
             fail("Should have been rejected.");
         } catch (RejectedExecutionException ex) {
             assertEquals("Got wrong RejectedExecutionException",
-                    "Maximum capacity (100) of tasks reached, for scheduled executor (foobar). "
+                    "Maximum capacity (100) of tasks reached for this member and scheduled executor (foobar). "
+                            + "Reminder, that tasks must be disposed if not needed.", ex.getMessage());
+        }
+    }
+
+    @Test
+    public void capacity_whenDefault_perPartition() {
+        String schedulerName = "foobar";
+        ScheduledExecutorConfig sec = new ScheduledExecutorConfig()
+                .setName(schedulerName)
+                .setCapacityPolicy(ScheduledExecutorConfig.CapacityPolicy.PER_PARTITION);
+
+        Config config = new Config().addScheduledExecutorConfig(sec);
+
+        HazelcastInstance[] instances = createClusterWithCount(1, config);
+        IScheduledExecutorService service = instances[0].getScheduledExecutorService(schedulerName);
+        String keyOwner = "hitSamePartitionToCheckCapacity";
+
+        for (int i = 0; i < 100; i++) {
+            service.scheduleOnKeyOwner(new PlainCallableTask(), keyOwner, 0, TimeUnit.SECONDS);
+        }
+
+        try {
+            service.scheduleOnKeyOwner(new PlainCallableTask(), keyOwner, 0, TimeUnit.SECONDS);
+            fail("Should have been rejected.");
+        } catch (RejectedExecutionException ex) {
+            assertEquals("Got wrong RejectedExecutionException",
+                    "Maximum capacity (100) of tasks reached for partition (262) and scheduled executor (foobar). "
```

Comment:
What about replacing the constant 262 with something like `getNodeEngine().getPartitionService().getPartitionId(keyOwner)`?